### PR TITLE
feat(): npm deprecate

### PR DESCRIPTION
**Pull request by Izak88**

### CODE REVIEW

File: tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { createPackageJson } from '../../utils/utils';
+
+export default function() {
+  return Promise.resolve()
+    .then(() => createPackageJson('package.json', 'test-package', '0.0.1'))
+    .then(() => npmLogin('admin', 'blabla', 'foo@bar.com'))
+    .then(() => npmPublish())
+    .then(() => executeSilent('npm logout'))
+    .then(() => deprecate('test-package', 'deprecate message'))
+    .then(res => {
```
Comment (jkuri) [inline tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts]: can you add `npm install test-package` after that command and check if returned deprecate message is same as you setted it before (deprecate message)?

Comment (jkuri) [reply tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts]: I think then this two tests would be perfect. Maybe you could also add `npm deprecate` test for scoped packages?

**Review by jkuri — CHANGES_REQUESTED** [review]
lets test this also for `@scoped` packages.

File: tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts (hunk 1 above)
Comment (Izak88) [reply tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts]: Sure, I'll add it...

**Comment by Izak88** [comment]
Changes after code review done.

File: tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts (hunk 1 above)
Comment (jkuri) [reply tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts]: here can easily be oneliner:
```ts
.then(res => res.code !== 0 ? Promise.resolve() : Promise.reject())
```

Comment (Izak88) [reply tests/e2e/tests/070_deprecate/010_unauth_deprecate.ts]: done

**Review by jkuri — APPROVED** [review]

**Pull request merged**